### ISSUE TITLE
Unsafe `Send` and `Sync` implementations

### ISSUE THREAD

**Issue by vladbat00** [naia-lib/naia#154]
The library marks a lot of types that are explicitly `!Send` and `!Sync` (meaning they can't be shared between threads) with `unsafe impl Send` and `unsafe impl Sync`. Considering that these types are actively used in the context of Bevy systems and wrapped into resources accessed with `Res`/`ResMut`, it's very likely to lead to UB.

Unsafe types in question:
https://github.com/search?q=repo%3Anaia-lib%2Fnaia+%22unsafe+impl%22&type=code

We should remove all the `unsafe impl Send` and `unsafe impl Sync` occurrences, even if it means that resources can no longer be accessed from different threads. Bevy supports `NonSend`, and I'd argue that limiting systems to run only on the main thread is a better compromise than causing UB.

**Comment by connorcarpenter** [comment]
I agree, thanks for the call-out @mvlabat  :+1: This may take a little bit for me to refactor though. :pray:

**Comment by connorcarpenter** [comment]
Done on `dev` in ba0185ce — every `unsafe impl Send` / `unsafe impl Sync` is now gone from the library.

Thanks for the report, @mvlabat. The good news is that the trade-off you anticipated turned out not to be necessary: none of this cost cross-thread access, and no Bevy resource had to move to `NonSend`.

I audited by experiment rather than by inspection — each impl was deleted and replaced with a `require_send::<T>()` static assertion, letting the compiler adjudicate rather than trusting the safety comments. Three groups came out of it:

**Genuinely unsound (2).** `NaiaServerError::Wrapped` held `Box<dyn Error>` and `NaiaClientError::Wrapped` held `Box<dyn Error + Send>`, with both `Send` *and* `Sync` asserted over them. `Wrapped` is a public variant, so a caller could store a payload satisfying neither; the server's own comment conceded this ("Callers who construct Wrapped with a !Send error type violate this invariant"), which is a documented hole rather than a safety argument. Both now hold `Box<dyn Error + Send + Sync>` and derive `Send`/`Sync` normally — the same idiom `NaiaServerSocketError` in this repo already used. This is breaking only for a caller wrapping an error that is not `Send + Sync`; every in-tree site wraps `std::io::Error` or a socket error.

**Stale (4).** `RecvState<E>` and `SendState<E>` are `Send` field-by-field. `SendState`'s comment blamed `Box<dyn PacketSender>`, but that trait has had `Send + Sync` supertraits for some time — the impl had outlived its reason. The `StepEntry<W>` types in the test and bench harnesses hold only a fn pointer, a `String` and a `Regex`, all already `Sync`.

**Deferred to upstream (2).** The `wasm_bindgen` `PacketSender` needed nothing: `web_sys::MessagePort` and `js_sys::Uint8Array` are themselves `Send + Sync` on `wasm32-unknown-unknown`. Deferring to that is also strictly safer than the hand-rolled version, because the upstream claim is target-gated and correctly stops applying if the target ever gains threads.

Static assertions were left in place of each removed impl so the compiler keeps proving these properties rather than a comment asserting them. Verified across the native workspace and both wasm backends (`wbindgen` and `mquad`): clippy clean, 138 suites / 830 tests passing.

The change is on `dev` and will ship in the next release.
